app/scripts/modules/core/src/pipeline/executions/executionGroup/ExecutionGroup.tsx

```diff
@@ -114,13 +118,21 @@ export class ExecutionGroup extends React.Component<IExecutionGroupProps, IExecu
   private startPipeline(command: IPipelineCommand): IPromise<void> {
     const { executionService } = ReactInjector;
     this.setState({ triggeringExecution: true });
-    return PipelineConfigService.triggerPipeline(
-      this.props.application.name,
-      command.pipelineName,
-      command.trigger,
-    ).then(
-      newPipelineId => {
-        const monitor = executionService.waitUntilNewTriggeredPipelineAppears(this.props.application, newPipelineId);
+
+    let triggerFunction: (a: string, b: string, c: any) => IPromise<string>;
+    let monitorFunction: (a: string) => IPromise<any>;
+    if (SETTINGS.feature.triggerViaEcho) {
+      triggerFunction = PipelineConfigService.triggerPipelineViaEcho;
```

Comment:
Please bind `PipelineConfigService` here and below, since someone could unwittingly use `this` inside `triggerViaEcho` or `triggerPipeline`.